ibm_cloud_sdk_core/jwt_token_manager.py

```diff
@@ -206,21 +67,15 @@ def _request(self,
                  data=None,
                  auth_tuple=None,
                  **kwargs) -> dict:
-        kwargs = dict({"timeout": 60}, **kwargs)
-        kwargs = dict(kwargs, **self.http_config)
-
-        if self.disable_ssl_verification:
-            kwargs['verify'] = False
+        return self._request_raw(method, url, headers=headers, params=params, data=data,
+                                 auth_tuple=auth_tuple, **kwargs).json()
```

Comment:
[_request](https://github.com/IBM/python-sdk-core/pull/72/files/146b4dc7f5adf7a612173ce4d41ed292035dae49#diff-4073e5945d5ebc449fcfce455620beccR61) was doing two things: making a prepared request via `requests.request` and a [json deserialization](https://github.com/IBM/python-sdk-core/pull/72/files/146b4dc7f5adf7a612173ce4d41ed292035dae49#diff-4073e5945d5ebc449fcfce455620beccL224). This deserialization hides information necessary to some alternative token managers - for example one that needs the cookies from the response so I've extracted [_request_raw](https://github.com/IBM/python-sdk-core/pull/72/files/146b4dc7f5adf7a612173ce4d41ed292035dae49#diff-5745e4f083a8d956453c9fe4e33e69bbR205) that only performs the request and exception throwing (so that is not lost, just moved to `TokenManager`).

Reply:
Please also take consider you can change the content of `request_token` functions in `IAMTokenManager` and in `CP4DTokenManager` to have `return response.json()` there.

Reply:
Or an other solution aspect `_request` would be good to keep in `TokenManager` level and in `JWTTokenManager` would better to have a `_jwt_request` and this is called from `IAMTokenManager` and `CP4DTokenManager`.

Reply:
> Please also take consider you can change the content of `request_token` functions in `IAMTokenManager` and in `CP4DTokenManager` to have `return response.json()` there.

`IAMTokenManager` and `CP4DTokenManager` (unchanged) use [JWTTokenManager's _request](https://github.com/IBM/python-sdk-core/pull/72/files/146b4dc7f5adf7a612173ce4d41ed292035dae49#diff-4073e5945d5ebc449fcfce455620beccR61) which does `.json()` so we should be okay.

Reply:
> Or an other solution aspect `_request` would be good to keep in `TokenManager` level and in `JWTTokenManager` would better to have a `_jwt_request` and this is called from `IAMTokenManager` and `CP4DTokenManager`.

Sorry, I don't understand. What would `_jwt_request` do?

Reply:
I think it would worth thinking about a function renaming `_request` -> `_jwt_request` and `_request_raw` -> `_request`.

Reply:
> `_request_raw` -> `_request`

That sounds like an API-breaking change as subclasses of JWTTokenManager (eg. [IAMTokenManager](https://github.com/IBM/python-sdk-core/blob/master/ibm_cloud_sdk_core/iam_token_manager.py#L109)) use 
 JWTTokenManager's _request and they expect a `.json()`-ed value.
I can do the follow-up this repo, but all external client code that inherit from `python-sdk-core`'s JWTTokenManager would need migration.

Reply:
While I like @vmatyusGitHub's suggested renaming, I think we should avoid API-breaking changes, so my preference is to leave the names as is.

Reply:
Actually, as I think about this more, I think it would be my preference to leave the request code here as is and simply leave it to other subclasses to implement their own request method.  Token manager really has no need for its own request method -- it doesn't use it for anything. So let subclasses implement this as they choose.

Reply:
I suppose the code is fine as is but let me throw out another alternative:
- implement `_request` in `JWTTokenManager` as it had been -- without calling `_request_raw`.
- Change `_request_raw` to just `_request` in `TokenManager` leaving the rest of the implementation as is.

This does duplicate some functionality, but sets us up for an easy transition in the next major release where we'd switch `JWTTokenManager` to use `TokenManager`'s `_request` and eliminate that duplicate functionality.  So we endure a little duplication now for the sake of compatibility but pave the way for a "clean" implementation later.

I think this suggestion makes the transition easier because subclasses of `TokenManager` can use `_request` now and won't have to change in the next major release.  (I suppose we could just continue to use `_request_raw` in the next major release, but I find that unappealing.)

How does this sound?

Reply:
I've just pushed f6ffb41 . Is this what you've meant?